python/ray/serve/batching.py

```diff
@@ -228,13 +253,18 @@ async def _process_batches(self, func: Callable) -> None:
                     results = await func_future
                     self._validate_results(results, len(batch))
                     for result, future in zip(results, futures):
-                        # If the client has disconnected, the future is canceled.
-                        # We should only set result when the connection is still live.
-                        if not future.cancelled():
-                            future.set_result(result)
+                        _set_result_if_not_done(future, result)
                 except Exception as e:
                     for future in futures:
-                        future.set_exception(e)
+                        _set_exception_if_not_done(future, e)
+        except Exception as e:
```

Comment:
Do we need 2 try-catch blocks stacked?

Reply:
Yeah, the inner block handles application errors, and the outer block handles system errors or bugs in Serve. We log an exception in the outer block that we shouldn't log for the inner block.